core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
+
+    path="WorldScreen/CityButton/DefenceTable",
+    defaultBgShape = BaseScreen.skinStrings.roundedTopEdgeRectangleSmallShape,
+    defaultBgBorder = BaseScreen.skinStrings.roundedTopEdgeRectangleSmallBorderShape) {
+
+    init {
+
+        val viewingCiv = UncivGame.Current.worldScreen!!.viewingCiv
+
+        borderSize = 4f
+        bgColor = Color.BLACK
+        bgBorderColor = when {
+            city.civInfo == viewingCiv -> colorFromRGB(255, 237, 200)
+            city.civInfo.isAtWarWith(viewingCiv) -> Color.RED
+            else -> Color.BLACK
+        }
+
+        pad(2f, 3f, 0f, 3f)
+
+        val cityStrength = CityCombatant(city).getDefendingStrength()
+        val cityStrengthLabel = "${Fonts.strength}$cityStrength"
+            .toLabel(fontSize = 12, alignment = Align.center)
+        add(cityStrengthLabel).colspan(2).grow().center()
+    }
+
 }
 
-class IconTable(borderColor: Color, innerColor: Color, borderSize: Float, borderOnTop:Boolean=true): BorderedTable(
+class AirUnitTable(city: City, numberOfUnits: Int, size: Float=14f) : BorderedTable(
+    path="WorldScreen/CityButton/AirUnitTable",
+    defaultBgShape = BaseScreen.skinStrings.roundedEdgeRectangleSmallShape,
+    defaultBgBorder = BaseScreen.skinStrings.roundedEdgeRectangleSmallShape) {
+
+    init {
+
+        padTop(2f)
+        padBottom(2f)
+
+        padLeft(10f)
+        padRight(10f)
+
+        val textColor = city.civInfo.nation.getInnerColor()
+
+        bgColor = city.civInfo.nation.getOuterColor()
+        bgBorderColor = city.civInfo.nation.getOuterColor()
+
+        val aircraftImage = ImageGetter.getImage("OtherIcons/Aircraft")
+        aircraftImage.color = textColor
+        aircraftImage.setSize(size, size)
+
+        add(aircraftImage)
+        add(numberOfUnits.toString().toLabel(textColor, size.toInt()))
+    }
+
+}
+
+private class StatusTable(city: City) : Table() {
+
+    init {
+
+        val viewingCiv = UncivGame.Current.worldScreen!!.viewingCiv
+
+        if (city.civInfo == viewingCiv && city.isConnectedToCapital() && !city.isCapital()) {
+            val connectionImage = ImageGetter.getStatIcon("CityConnection")
+            add(connectionImage).size(18f)
+        }
+
+        if (city.isInResistance()) {
+            val resistanceImage = ImageGetter.getImage("StatIcons/Resistance")
+            add(resistanceImage).size(18f).padLeft(2f)
+        }
+
+        if (city.isPuppet) {
+            val puppetImage = ImageGetter.getImage("OtherIcons/Puppet")
+            add(puppetImage).size(18f).padLeft(2f)
+        }
+
+        if (city.isBeingRazed) {
+            val fireImage = ImageGetter.getImage("OtherIcons/Fire")
+            add(fireImage).size(18f).padLeft(2f)
+        }
+
+        if (city.civInfo == viewingCiv && city.isWeLoveTheKingDayActive()) {
+            val wltkdImage = ImageGetter.getImage("OtherIcons/WLTKD")
+            add(wltkdImage).size(18f).padLeft(2f)
+        }
```

Comment:
Lots of `size(18f).padLeft(2f)` that seems like it should be moved to a constant somewhere.